src/main/java/cz/tstrecha/timetracker/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package cz.tstrecha.timetracker.service.impl;
+
+import cz.tstrecha.timetracker.constant.IdentifierType;
+import cz.tstrecha.timetracker.dto.LoggedUser;
+import cz.tstrecha.timetracker.dto.TaskCreateRequestDTO;
+import cz.tstrecha.timetracker.dto.TaskDTO;
+import cz.tstrecha.timetracker.dto.mapper.TaskMapper;
+import cz.tstrecha.timetracker.repository.TaskRepository;
+import cz.tstrecha.timetracker.service.TaskService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@RequiredArgsConstructor
+public class TaskServiceImpl implements TaskService {
+
+    private final TaskRepository taskRepository;
+
+    private final TaskMapper taskMapper;
+
+    public TaskDTO createTask(TaskCreateRequestDTO taskRequest, LoggedUser loggedUser) {
+        var taskEntity = taskMapper.fromRequest(taskRequest, loggedUser.getUserEntity());
+        taskEntity = taskRepository.save(taskEntity);
+        return taskMapper.toDTO(taskEntity);
+    }
+
+    public TaskDTO createEmptyTask(IdentifierType identifierType, String identifierValue, LoggedUser loggedUser) {
+        var request = new TaskCreateRequestDTO();
```

Comment:
This is not wrong, but..

You sholdn't craete a request just to map it into a entity -> you can just create an entity, set the params using switch below and save it, then map to dto and return the dto, this use of mapper is unnecessary